### DISCUSSION TITLE
ring form to kekulized form

### DISCUSSION THREAD

**Discussion by shyamramesh** [rdkit/rdkit#6782] [Q&A] [answered]
<img width="406" alt="image" src="https://github.com/rdkit/rdkit/assets/8455786/fca44ba9-fbc7-4a8e-89f8-161510907b2c">

Is it possible to **convert ring form to kekulized form and viceversa** with RDKitJS?
   function toggleRing(){
   var smi='O=C([O-])C1=CN=CN=C1';
   var prefs = {sanitize:false,kekulize: false};
   var mol = RDKit.get_mol(res,JSON.stringify(prefs));
   **mol.convert_to_aromatic_form();**
   //get smiles aromatic form
   var aromatic_smiles=mol.get_smiles();
   var amol=RDKit.get_mol(aromatic_smiles,JSON.stringify(prefs));
   **var dopt['prepareMolsBeforeDrawing']=false;**
   svg= amol.get_svg_with_highlights(JSON.stringify(dopt)) -----> this is not giving ring form svg
   
   how to get ring form in RDKitjs?

**Comment by ptosco** [comment, marked as answer]
@shyamramesh This will do what you describe:
```
const aromatic_div = document.getElementById('aromatic_form');
if (aromatic_div) {
    const smi = 'O=C([O-])C1=CN=CN=C1';
    let mol;
    try {
        mol = rdkitModule.get_mol(smi);
        aromatic_div.innerHTML = mol.get_svg_with_highlights(JSON.stringify({kekulize: false}));
    } finally {
        if (mol) {
            mol.delete();
        }
    }
}
```
Note that you will not get the circle representation that you exemplify above; you will rather get dotted lines:
![image](https://github.com/rdkit/rdkit/assets/5244385/c92f26a5-9d49-49a0-8278-e7cfc8d0c7c1)

**Reply by shyamramesh** [reply]
Thanks a lot.It worked
